### PR TITLE
CARTO: fetchMap support spatial index layers

### PR DESCRIPTION
**Pull request by felixpalmer**
For https://github.com/visgl/deck.gl/issues/6912

#### Change List
- Unify layer selection based on tileset definition between `CartoLayer` & `fetchMap`
- Generate spatial index tile layers in fetchMap
- Map aggregation parameters from layer style
- Support stying for dynamic tiling layers via the stats API
- Tests

### CODE REVIEW

**Review by padawannn — CHANGES_REQUESTED** [review]
@felixpalmer I think that you're not building the `aggregationExp` from the layer style? Have you tried to style a map by a property in builder and check if it works with the fetchMap?

**Comment by felixpalmer** [comment]
> `aggregationExp`

@padawannn No, you're right - that is missing. Could you point me at where that is done in builder? I can't seem to find the code

**Review by alasarr — DISMISSED** [review]

File: modules/carto/src/api/layer-map.ts [hunk 1]
```diff
@@ -163,17 +165,36 @@ export function getLayer(
   };
 }
 
+export function layerFromTileDataset(
```
Comment (alasarr) [inline modules/carto/src/api/layer-map.ts:168]: 👍

**Comment by alasarr** [comment]
>@padawannn No, you're right - that is missing. Could you point me at where that is done in builder? I can't seem to find the code

You can try at dedicated server 05, click on layer and fill by color based on a property

**Review by padawannn — CHANGES_REQUESTED** [review]

**Review by padawannn — APPROVED** [review]

**Pull request merged**